assets/src/edit-story/app/prepublish/getPrepublishErrors.js

```diff
@@ -120,7 +120,54 @@ function getPrepublishErrors(story) {
       return [...storyGuidance, ...pageGuidance, ...elementGuidance];
     })
     .flat();
-  return checklistResult;
+
+  const asyncResult = await getPrepublishErrorsAsync(story);
+
+  return [...checklistResult, ...asyncResult];
+}
+
+async function getPrepublishErrorsAsync(story) {
```

Comment:
At first glance I find it a bit odd to have `getPrepublishErrors()` and `getPrepublishErrorsAsync()`, and `warnings and `asyncWarnings`. Especcially since the former two are both `async` functions. Doesn't seem like a very dev-friendly API. Can't everything be async and unified in 1 place?

Reply:
I can refactor, I was working on the image stuff for a long time so I wanted to be able to follow the flow without getting myself confused.

Reply:
The main thing was I didn't want to rewrite the reduce or change the data structures. But spreading the guidance message doesn't let me resolve promises with `promise.all` very easily.

Reply:
I think what's there now is much better :) thank you